### PR TITLE
XS✔ ◾ Do you know the best practices for frontmatter in markdown

### PR DESCRIPTION
**Pull request by ozairashfaqueSSW**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

Email: 2-Month Employment Review - Ozair Ashfaque
**Task assigned for 4-Month Review:** 
-   Check with @wicksipedia to discuss adding frontmatter rules to the Gatsby website

> 2. What was changed?

- Created rule **Do you know the best practices for frontmatter in markdown** with the help of @Marxoz 
- Added more points for better frontmatter in markdown
- Added rule in rules-to-better-websites-development category
- Added Matt Wicks for author of rule

> 3. Did you do pair or mob programming?

No

### CODE REVIEW

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 1]
```diff
@@ -97,3 +99,124 @@ authors:
...
+```
+---
+t: My Article
+auth: John Doe
+---
+```
+:::
+ 
+::: bad 
+Figure: Bad Example {Shortened or unclear keys can lead to confusion.}
```
Comment (tiagov8) [inline rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md:125]: Hi @ozairashfaqueSSW 

Unclear about the use of { ... } on your captions...

Comment (ozairashfaqueSSW) [reply rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md]: Hi @tiagov8 , 
I have fixed the format. 
Please review again

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 2]
```diff
@@ -95,5 +97,126 @@ authors:
...
+Figure: Bad Example of using inline HTML can cause unexpected rendering or parsing issues.
+:::
+ 
+Stick to plain markdown:
+ 
+::: greybox
+```
+---
+title: My Article
+author: John Doe
```
Comment (wicksipedia) [inline rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md:213]: this is the same example as the meaningful keys one. I think this should be an example with more detail e.g. more complex properties

add a number, boolean, array, and maybe an object following all the rules

**Review by wicksipedia — DISMISSED** [review]

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md (hunk 2 above)
Comment (ozairashfaqueSSW) [reply rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md]: Example I have added :
image

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 3]
```diff
@@ -37,7 +39,7 @@ title+author: My Article by John
 :::
 
 ::: bad 
-Figure: Bad Example {Non-standard practices can lead to parsing errors and inconsistencies.} 
+Figure: Bad Example of non-standard practices can lead to parsing errors and inconsistencies.
```
[the remark this answers is not in the document]
Comment (ozairashfaqueSSW) [reply rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md]: Removed greyboxes

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 4]
```diff
@@ -17,7 +19,7 @@ However, when not structured properly, it can lead to parsing errors, inconsiste
 
 <!--endintro-->
 
-#### Key-Value Pairs
+#### Key-Value Pair Organization
 
 Frontmatter in markdown primarily utilizes key-value pairs to represent metadata. Each key represents an attribute, like the title or the author, and the value associated with it provides specific information related to that attribute.
```
[the remark this answers is not in the document]
Comment (ozairashfaqueSSW) [reply rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md]: Fixed the intro

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 5]
```diff
@@ -17,41 +19,37 @@ However, when not structured properly, it can lead to parsing errors, inconsiste
...
 –––
 title+author: My Article by John
 2023-10-31
 –––
 ```
-:::
 
 ::: bad 
-Figure: Bad Example {Non-standard practices can lead to parsing errors and inconsistencies.} 
+❌Figure: Bad Example of non-standard practices can lead to parsing errors and inconsistencies.
```
Comment (wicksipedia) [inline rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md:40]:
```suggestion
Figure: Bad Example of non-standard practices can lead to parsing errors and inconsistencies.
```

bad/good examples have classes that apply ticks/crosses automatically

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 6]
```diff
@@ -17,41 +19,37 @@ However, when not structured properly, it can lead to parsing errors, inconsiste
...
-author: John Doe
+author: Bob Northwind
 date: 2023-10-31
 –––
 ```
-:::
 
 ::: good 
-Figure: Good Example {Clear key-value pairs make it easy to understand and extract the metadata.} 
+✅Figure: Good Example of clear key-value pairs make it easy to understand and extract the metadata.
```
Comment (wicksipedia) [inline rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md:52]:
```suggestion
Figure: Good Example - key-value pairs make it easy to understand and extract the metadata.
```

bad/good examples have classes that apply ticks/crosses automatically

Comment (ozairashfaqueSSW) [reply rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md:52]: I will fix it now

File: rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md [hunk 7]
```diff
@@ -97,3 +99,124 @@ authors:
...
+#### Use Meaningful Keys
+ 
+The keys you choose for your Frontmatter should be meaningful and descriptive. They act as identifiers for the associated values, so it's essential that they clearly convey the data they represent.
+ 
+- **Descriptive Names:** Instead of using `desc`, use `description`. Instead of `auth`, use `author`.
+- **Consistency:** Stick to a consistent naming convention, whether it's camelCase, snake_case, or kebab-case.
+ 
+Avoid non-descriptive keys:
+ 
+::: greybox
```
[the remark this answers is not in the document]
Comment (tiagov8) [reply rules/do-you-know-the-best-practices-for-frontmatter-in-markdown/rule.md]: Hey @ozairashfaqueSSW I've done task 2 for you ;)

**Review by tiagov8 — APPROVED** [review]

**Review by tiagov8 — APPROVED** [review]

**Pull request merged**